activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -749,9 +749,34 @@ def configure_connection
           if @config[:timeout] && @config[:retries]
             raise ArgumentError, "Cannot specify both timeout and retries arguments"
           elsif @config[:timeout]
-            @raw_connection.busy_timeout(self.class.type_cast_config_to_integer(@config[:timeout]))
+            timeout = self.class.type_cast_config_to_integer(@config[:timeout])
+            raise TypeError, "Timeout must be an integer" unless timeout.is_a?(Integer)
+
+            timeout_seconds = timeout.fdiv(1000)
+            retry_interval = 6e-5 # 60 microseconds
```

Comment:
How was this value selected?

Reply:
Loosely and quickly 😅

I wanted a value that would allow for sub-millisecond resolution, since SQLite can perform various operation in microseconds (https://www.sqlite.org/fasterthanfs.html#approx, https://github.com/chrisdavies/dbench) but also wouldn't create too much thrashing (busy looping as you say). 60 seconds in a minute, 60 minutes in an hour. 60 just seemed like a place to start, and when running the code, I went from errors all the time to never errors, so I've kept it.

I opened the PR looking to get wiser insight from people like you, as while I have a clear picture of the shape of the problem and how I can generally fix it, I am in no way certain that this is the perfect or even one of the better solutions. I'm very much open to finding a better solution to the problem.